test/integration/tas/tas_test.go

```diff
@@ -839,3 +840,56 @@ var _ = ginkgo.Describe("Topology Aware Scheduling", ginkgo.Ordered, func() {
 		})
 	})
 })
+
+var _ = ginkgo.Describe("Topology validations", func() {
+	ginkgo.When("Creating a Topology", func() {
+		ginkgo.DescribeTable("Validate Topology on creation", func(topology *kueuealpha.Topology, matcher types.GomegaMatcher) {
+			err := k8sClient.Create(ctx, topology)
+			if err == nil {
+				defer func() {
```

Comment:
Similarly here, we can move defer after we check there is no error

Reply:
Do you request all defer block moving?

https://github.com/search?q=repo%3Akubernetes-sigs%2Fkueue+defer+func%28%29+%7B+path%3A%2F%5Etest%5C%2Fintegration%5C%2F%2F&type=code

Reply:
I see, I didn't see that, let's do it in a follow up.

Reply:
Additionally, I think the current defer place would be better. Because if there are any panic in the `gomega.Expect`, the defer will never performed, right?

Reply:
I see, yes, I didn't think about it. However, it seems this pattern is quite verbose. Also, if there is a panic we probably don't want to continue tests anyway, so cleanup might not be necessary I think.

Reply:
Another pattern would be to use AfterEach as in most tests, and cleanup there, so I think could re-write this verbose pattern, but I'm ok as is since this is consitent with other tests in this file.

Reply:
> Another pattern would be to use AfterEach as in most tests, and cleanup there, so I think could re-write this verbose pattern,

IIUC, the solution is not available based on the DecribeTable specifications. If we use the AfterEach, the cleanup will be performed when all Entry completed. But we want to perform the cleanup when each Entry.

Reply:
Hm, in that case I would suggest to not use `DecribeTable` or just not support cleanup on panics. I think in case of panic any method will not fully ok. Imagine you have After each with two cleanup statements. If first fails we don't cleanup the second anyway, so I wouldn't worry about them too much.

Reply:
> Hm, in that case I would suggest to not use DecribeTable or just not support cleanup on panics.

If we migrate to the `ginkgo.It`, it will bring us huge testing codes...
So, the current `defer` solution is not bad one. But I agree with not good solution.